src/NuGet.Core/NuGet.Packaging/Signing/Verification/CertificateHashAllowListEntry.cs

```diff
@@ -14,8 +17,65 @@ public class CertificateHashAllowListEntry : VerificationAllowListEntry
         public CertificateHashAllowListEntry(VerificationTarget target, SignaturePlacement placement, string fingerprint, HashAlgorithmName algorithm)
             : base(target, placement)
         {
-            Fingerprint = fingerprint;
+            if (!Enum.IsDefined(typeof(SignaturePlacement), placement))
```

Comment:
why?
How can this happen?
There's no serialization that could break it.

Reply:
There is no case where this will break, I added this for the sake of the sanity of the API itself, in case there is an external user to not allow it to break. Do you think it should be deleted?

Reply:
No, not necessarily, it doesn't hurt, but I also don't think it has to be there. 

We've normally only done this in plugin code, but if you feel strongly about it, keep it :)

Reply:
We also have this in other verification settings, which is where I got it from.